server/views/pages/goal/update/index.njk

```diff
@@ -0,0 +1,161 @@
+{% extends "../../../partials/layout.njk" %}
+
+{% set pageId = "update-goal" %}
+{% set pageTitle = applicationName + " - Home" %}
+
+{% block beforeContent %}
+  {{ govukBreadcrumbs({
+    items: [
+      {
+        text: "Digital Prison Services",
+        href: dpsUrl
+      },
+      {
+        text: "Manage learning and work progress",
+        href: "/"
+      },
+      {
+        text: "Manage " + prisonerSummary.firstName | capitalize + " " + prisonerSummary.lastName | capitalize + "'s learning plan",
+        href: "/plan/" + prisonerSummary.prisonNumber + "/view/overview"
+      }
+    ],
+    classes: 'govuk-!-display-none-print'
+  }) }}
+{% endblock %}
+
+{% block content %}
+
+  {% if errors.length > 0 %}
+    {{ govukErrorSummary({
+      titleText: 'There is a problem',
+      errorList: errors,
+      attributes: { 'data-qa-errors': true }
+    }) }}
+  {% endif %}
+
+  {% include "../../../partials/prisonerBanner.njk" %}
+
+  <div class="govuk-grid-row">
+    <div class="govuk-grid-column-two-thirds">
+
+      <h1 class="govuk-heading-l">You are updating a goal for {{ prisonerSummary.firstName | capitalize }} {{ prisonerSummary.lastName | capitalize }}</h1>
+
+        <form class="form" method="post" novalidate="">
+            <input type="hidden" name="_csrf" value="{{ csrfToken }}" />
+            <input type="hidden" name="prisonNumber" value="{{ form.prisonNumber }}" />
+            <input type="hidden" name="goalReference" value="{{ form.goalReference }}" />
+
+            {{ govukTextarea({
+              name: "description",
+              id: "description",
+              label: {
+                text: "Description",
+                classes: "govuk-label--m"
+              },
+              value: form.title,
+              errorMessage: errors | findError('description')
+            }) }}
+
+            {{ govukTextarea({
+              name: "note",
+              id: "note",
+              label: {
+                text: "Add or edit note",
+                classes: "govuk-label--m"
+              },
+              value: form.note,
+              errorMessage: errors | findError('note')
+            }) }}
+
+            <h2 class="govuk-heading-m">Edit and remove steps</h2>
+
+            {% for step in form.steps %}
+              <input type="hidden" name="stepReference" value="{{ step.reference }}" />
+
+              {{ govukTextarea({
+                name: 'title-' + step.reference,
```

Comment:
I think the `name` and `id` attributes will need to reference the form structure, and in particular the fact that the `steps` field is an array
I think it'll need to be something like:
```
name: 'steps[' + loop.index + '].title',
id: 'steps[' + loop.index + '].title',
```
Something like that (assuming we can get the loop index?)

You'll need to do this on all the step fields, so that when the form gets `POST`ed back we get rehydrate the form data into the `UpdateGoalForm` object

Reply:
Just noticed that too from your PR https://github.com/ministryofjustice/hmpps-education-and-work-plan-ui/pull/87#discussion_r1283193701

Will update to match